README.md

```diff
@@ -20,12 +20,36 @@ Any exceptions to this are noted in the individual `README.md` files.
 
 ## Installation
 
-TBD
+The repository layout structure is as defined by [the specification](https://github.com/open-telemetry/opentelemetry-specification/blob/master/specification/library-layout.md). The `api`
+directory includes header-only abstract classes and no-op implementations that comprise the OpenTelemetry API following
+[the
+specification](https://github.com/open-telemetry/opentelemetry-specification).
+The `sdk` folder is the reference implementation of the API.
+
+The library would be distributed in source-only format. No shared/static library or distribution package (rpm, deb etc) would be made available as part of Release. See [RELEASING.md](./RELEASING.md).
+
+This project supports `Bazel` and `CMake` builds.
+
+### Using CMake
+On most platforms with all dependencies installed, the following command will compile the source files, and install the header files:
 
-## Quick Start
+```sh
+cmake -Bcmake-out
+cmake --build cmake-out
+sudo cmake --build cmake-out --target install
 
+#To run the tests
+cd cmake-out && ctest
```

Comment:
To elaborate a little bit on this, ideally - the local dev build process could be as easy as:

- `tools/build.cmd` - on Windows, which will auto-detect the compiler AND setup all the build tools needed (unless you want to explicitly specify the compiler, for that we'll have `tools/build-vs201x.cmd` )

- `tools/build.sh` - for anything POSIX that only has `/bin/bash` : Linux, Mac, Android.

And

- crossplat `vcpkg install opentelemetry[:feature_pack]`

Maybe we can remove the cmake section here? And I can elaborate on a separate doc with the tooling I'm adding, as t's near-zero effort to onboard. I already added some drafts for `vcpkg` build process in the other PR.

Reply:
sure, I can remove the steps from here, and instead add one-liner saying Refer to (say) INSTALL.md for details.

Reply:
I can post the recommendations in `INSTALL.md`, to cover various  ways to build (or build from source and install) instructions. Separate sections for common build options:
- CMake
- Bazel
- vcpkg
- Visual Studio 2019 with CMake support (GUI build, no command-line tooling)
- command line wrappers to facilitate the build
- docker cross-plat builds

I'll send a separate PR with the doc. Most of it is already implemented in #377

Reply:
It's just important that we have detailed CMake documentation somewhere, be it here on in the `INSTALL.md`. It should also cover the most useful build flags, like `WITH_OTLP`, `WITH_TESTS` and `WITH_EXAMPLES`.